[BUG] Levant does not handle periodic job types properly

While the `service` scheduler is the only one to properly support deployments, there is value in using levant to render job file templates, it should therefore support registering other scheduler type jobs.

Currently solved with the following:
```sh
levant render -var x=y job.nomad > rendered.nomad
nomad run rendered.nomad
```

Actually I should have clarified that I've noticed this problem was with jobs of type `periodic`, I've assumed it would affect all jobs but it actually seems to be due to there not being an evaluation:

```
[INFO] levant/deploy: using dynamic count 1 for job backup-bot and group backup-bot
[INFO] levant/deploy: triggering a deployment of job backup-bot
[ERROR] levant/deploy: Unexpected response code: 500 (rpc error: eval lookup failed: index error: UUID must be 36 characters)
[ERROR] levant/command: deployment of job backup-bot failed
```

Can probably be fixed by just returning if there is no evaluation.

--- a/levant/deploy.go
+++ b/levant/deploy.go
@@ -78,6 +78,12 @@ func (c *nomadClient) Deploy(job *nomad.Job, autoPromote int, forceCount bool) (
 		return
 	}
 
+	// GH-50: batch job types do not return an evaluation upon registration.
+	if eval == nil && *job.Type == nomadStructs.JobTypeBatch {
+		logging.Debug("levant/deploy: job type %s does not create evaluations", nomadStructs.JobTypeBatch)
+		return true
+	}
+
 	// Trigger the evaluationInspector to identify any potential errors in the
 	// Nomad evaluation run. As far as I can tell from testing; a single alloc
 	// failure in an evaluation means no allocs will be placed so we exit here.

